lib/tasks/cleanup.rake

```diff
@@ -0,0 +1,22 @@
+# typed: false
+# frozen_string_literal: true
+
+namespace :cleanup do
+  desc 'Remove unattached files'
+  task uploads: :environment do
+    # Retrieve all blob keys that were created 7 days ago and do not have attachments
+    unattached_blobs = ActiveStorage::Blob.left_outer_joins(:attachments)
+                                          .where('DATE(active_storage_blobs.created_at) = ?', 7.days.ago)
+                                          .where(attachments: { id: nil })
+                                          .pluck(:id, :key)
+
+    # Remove the blob and the file from the disk
+    unattached_blobs.each do |blob_id, key|
+      ActiveStorage::Blob.find(blob_id)
+      uploaded_file_path = File.join(Settings.file_uploads_root, key[0, 2], key[2, 2], key)
```

Comment:
I'm pretty sure you can do `blob.purge_later` and then you don't have to worry about all this file stuff.

Reply:
https://github.com/rails/rails/blob/328113b0adfb516639ea0ad9845e15877b8c8141/activestorage/app/models/active_storage/blob.rb#L303

Reply:
Good call, that simplified this into a one liner.